core/black_white_proxy.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+# coding=utf-8
+
+import re
+from PyQt5.QtNetwork import QNetworkProxy, QNetworkProxyFactory
+from core.utils import get_emacs_buffer_url
+
+class BlackWhiteQNetworkProxyFactory(QNetworkProxyFactory):
+    """
+    Proxy factory that enables non-default proxy list when
+    requested URL is matched by one of whitelist patterns
+    while not being matched by one of the blacklist patterns.
+    """
+    def __init__(self, proxy=None, blacklist=None, whitelist=None):
+        self.proxy = proxy or None
+        self.blacklist = blacklist or []
+        self.whitelist = whitelist or []
+
+        super(BlackWhiteQNetworkProxyFactory, self).__init__()
+
+    def queryProxy(self, query=None, *args, **kwargs):
+        url = str(get_emacs_buffer_url())
```

Comment:
This is not right. 

Please reference https://github.com/yeyanchao/calibre/blob/e4d06dd27cd4d9139f5acaabda6e11804ab4734b/src/calibre/web/jsbrowser/browser.py

or 

https://github.com/spyder-ide/conda-manager/blob/c5dd8963490defc0241eebc41a66f034a9821e1f/conda_manager/api/download_api.py